nemo/lightning/megatron_parallel.py

```diff
@@ -1711,7 +1711,10 @@ def masked_token_loss(tensor: Tensor, mask: Tensor):
     """
     losses = tensor.float()
     loss_mask = mask.view(-1).float()
-    loss = torch.sum(losses.view(-1) * loss_mask) / loss_mask.sum()  # sequence level nll
+    num_valid_tokens = loss_mask.sum()
+    if num_valid_tokens < 0.5:  # no valid tokens
```

Comment:
can you explain a bit more when this is the case? is this only valid for neva?

Reply:
not only for neva, also for SFT. If the system and user prompt is very long, and predict answer only. After truncation from right, there might not be any answer/valid tokens.

Reply:
Usually we truncate the input/context and keep answer intact, so that wouldn't happen

Reply:
yep the logic is a bit different for vlm. We don't want to truncate from left.